src/Uno.UWP/UI/Core/CorePhysicalKeyStatus.cs

```diff
@@ -10,5 +12,27 @@ public partial struct CorePhysicalKeyStatus
 		public bool IsMenuKeyDown;
 		public bool WasKeyDown;
 		public bool IsKeyReleased;
+
+		public bool Equals(CorePhysicalKeyStatus other) =>
+			RepeatCount == other.RepeatCount && ScanCode == other.ScanCode &&
+			IsExtendedKey == other.IsExtendedKey && IsMenuKeyDown == other.IsMenuKeyDown &&
+			WasKeyDown == other.WasKeyDown && IsKeyReleased == other.IsKeyReleased;
+
+		public override bool Equals(object? obj) => obj is CorePhysicalKeyStatus other && Equals(other);
+
+		public override int GetHashCode() =>
+			new
+			{
+				RepeatCount,
+				ScanCode,
+				IsExtendedKey,
+				IsMenuKeyDown,
+				WasKeyDown,
+				IsKeyReleased
+			}.GetHashCode();
```

Comment:
This is interesting, I have never thought of using anonymous types that way.

Though this is not efficient and allocates unnecessarily.

```csharp
		public override int GetHashCode()
		{
			var hashCode = -1614557298;
			hashCode = hashCode * -1521134295 + RepeatCount.GetHashCode();
			hashCode = hashCode * -1521134295 + ScanCode.GetHashCode();
			hashCode = hashCode * -1521134295 + IsExtendedKey.GetHashCode();
			hashCode = hashCode * -1521134295 + IsMenuKeyDown.GetHashCode();
			hashCode = hashCode * -1521134295 + WasKeyDown.GetHashCode();
			hashCode = hashCode * -1521134295 + IsKeyReleased.GetHashCode();
			return hashCode;
		}
```